payload/payload/CourseManager.cc

```diff
@@ -1070,6 +1071,40 @@ void *CourseManager::loadLocalizedFile(const char *prefix, const char *suffix, J
     return nullptr;
 }
 
+void *CourseManager::loadCourseFile(const char *zipPath, const char *filePath, u32 *size) const {
+    bool ok;
+    ZIPFile zipFile(zipPath, ok);
+    if (!ok) {
+        return nullptr;
+    }
+    return loadCourseFile(zipFile, filePath, size);
+}
```

Comment:
With a such a generic function signature, should this be ZIP-specific? Would we not want some kind of dispatcher for compression type when introducing SZS? I could also be missing the point.

Reply:
Custom tracks are always distributed as ZIP archives that contain an RARC archive named `track.arc` among other files. This PR adds SZS compression support for the inner archive (the base game supports it but doesn't use it, but some custom tracks do).

Reply:
Got it, makes sense to me.